samples/java-valueentity-shopping-cart/pom.xml

```diff
@@ -169,12 +169,21 @@
         <executions>
           <execution>
             <goals>
-              <goal>java</goal>
+              <goal>exec</goal>
             </goals>
           </execution>
         </executions>
         <configuration>
-          <mainClass>${mainClass}</mainClass>
+          <executable>java</executable>
```

Comment:
This doesn't pick up the `JAVA_HOME` directory as `exec:java` does (because it runs it in the same JVM basically). So, that might be a bit of a pain to use if you have multiple JDKs installed and the `java` in the `PATH` does not correspond to `JAVA_HOME`. Or should we say that's an oddity of the environment?

We could keep using `exec:java` with a configuration like this:

```xml
      <plugin>
        <groupId>org.codehaus.mojo</groupId>
        <artifactId>exec-maven-plugin</artifactId>
        <version>3.0.0</version>
        <executions>
          <execution>
            <goals>
              <goal>java</goal>
            </goals>
          </execution>
        </executions>
        <configuration>
          <mainClass>${mainClass}</mainClass>
          <systemProperties>
            <systemProperty>
              <key>akkaserverless.user-function-interface</key>
              <value>0.0.0.0</value>
            </systemProperty>
          </systemProperties>
        </configuration>
      </plugin>
```

Reply:
Aha, I didn't think about sysprops being mutable so thought that wouldn't work.

But, wouldn't you either have some additional proxy tool `java` on the `PATH` if you have many installations, which in turn looks for `JAVA_HOME` or if you use jabba/sdkman whatnot select with that and get a java on `PATH`? (That's how it works out of the box with Mac at least).

Feels like a very cornery case env to me (but I might be wrong).

Reply:
Let's go with this for now and we can re-think if it turns out it is a common problem.

Reply:
:+1: yeah might be how I use my environment.